main/src/main/scala/sbt/Defaults.scala

```diff
@@ -3585,15 +3585,57 @@ object Classpaths {
   ): Initialize[Task[Unit]] =
     publishTask(config)
 
+  private def logSkipPublish(log: Logger, ref: ProjectRef): Unit =
+    log.debug(s"Skipping publish* for ${ref.project}")
+
+  private def checkAndPublishLocal(
+      currentCon: PublishConfiguration,
+      publishLocalConf: PublishConfiguration,
+      publishSkipLocal: Boolean,
+      module: IvySbt#Module,
+      log: Logger,
+      ref: ProjectRef
+  )(notPublishLocal: () => Unit): Unit = {
+    if (currentCon == publishLocalConf) {
```

Comment:
Is this equality comparison correct for checking if the current publish target is `publishLocal` or should something else be done?

Reply:
Ah yes that would be nicer, lets see what response I get from sbt maintainer to see if I am on write track and if so I can clean it up later

Reply:
So I just had a quick look at this and I am not sure if this is possible without breaking binary compatibility (which I think is a concern as they appear to be public functions?)

The first issue is that if I change `publishTask` to `
```scala
def publishTask(
    config: TaskKey[PublishConfiguration],
    skipPublish: TaskKey[Boolean]
)
```
Then this will conflict with an already existing deprecated method (i.e.

```scala
@deprecated("Use variant without delivery key", "1.1.1")
def publishTask(
    config: TaskKey[PublishConfiguration],
    deliverKey: TaskKey[_],
): Initialize[Task[Unit]] =
  publishTask(config)
```

Both methods will have the same type after erasure.

Instead you could define

```scala
def publishTask(
    config: TaskKey[PublishConfiguration],
    skipPublish: Boolean
)
```

To deal with the erasure problem but on the assumption we are dealing with public functions, such a function is out of place in terms of API design.

If I can get rid of the deprecated `publishTask` function then this makes perfect sense. @eed3si9n What do you think?

Reply:
I don't think we can remove methods during 1.x series.

Reply:
Okay so you fine leaving it as is?

Reply:
> Okay so you fine leaving the implementation it as is?

Yea.